### PR TITLE
MSAL Upgrade for Common Library

### PR DESCRIPTION
**Pull request by ozanovus**
**Task name**: azure-arm-rest-v2

**Description**: 

Structure update for new MSAL library. 

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) Y
N/A

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it

Major version change to prevent automatic Task updates.

- [X] Checked that applied changes work as expected

### CODE REVIEW

File: common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts [hunk 1]
```diff
@@ -62,124 +66,277 @@ export class ApplicationTokenCredentials {
...
             (error) => {
-                deferred.reject(error)
+                deferred.reject(error);
             }
         );
 
         return deferred.promise;
     }
 
+    public getDomain(): string {
```
Comment (starkmsu) [inline common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts:144]: Maybe rename it to getTenantId to be consistent?

File: common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts [hunk 2]
```diff
@@ -7,37 +7,42 @@ import constants = require('./constants');
...
+    private secret?: string;
+    private accessToken?: string;
+    private certFilePath?: string;
+    private isADFSEnabled?: boolean;
     private token_deferred: Q.Promise<string>;
+    private useMSAL: boolean;
+    private msalInstance: msal.ConfidentialClientApplication;
 
-    constructor(clientId: string, domain: string, secret: string, baseUrl: string, authorityUrl: string, activeDirectoryResourceId: string, isAzureStackEnvironment: boolean, scheme?: string, msiClientId?: string, authType?: string, certFilePath?: string, isADFSEnabled?: boolean, access_token?: string) {
+    constructor(clientId: string, tenantID: string, secret: string, baseUrl: string, authorityUrl: string, activeDirectoryResourceId: string, isAzureStackEnvironment: boolean, scheme?: string, msiClientId?: string, authType?: string, certFilePath?: string, isADFSEnabled?: boolean, access_token?: string, useMSAL?: boolean) {
```
Comment (starkmsu) [inline common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts:34]: Why no tenantId?

File: common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts [hunk 3]
```diff
@@ -62,124 +66,277 @@ export class ApplicationTokenCredentials {
...
+                break;
+            case AzureModels.Scheme.SPN:
+            default:
+                this.configureMSALWithSP(msalConfig);
+                break;
+        }
+    }
+
+    private configureMSALWithMSI(msalConfig: msal.Configuration) {
+        let resourceID = this.activeDirectoryResourceId;
```
Comment (starkmsu) [inline common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts:192]: resourceId?

File: common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts (hunk 2 above)
Comment (ozanovus) [reply common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts]: AzureModel file contains lots of "ID" definition, that's why sometimes I've used ID syntax but for the file consistency I've changed them to "Id"

**Review by starkmsu — APPROVED** [review]

**Pull request merged**